rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -63,4 +78,95 @@ class CostAccountingSpec extends FlatSpec with TripleEqualsSupport {
 
     assertOutOfPhloError(test)
   }
+
+}
+
+class CostAccountingSpec extends FlatSpec with Matchers with PropertyChecks {
+
+  private[this] def evaluateWithCostLog(
+      initialPhlo: Long,
+      contract: String
+  ) = {
+    implicit val errorLog = new ErrorLog[Task]()
+    implicit val costAlg: CostAccounting[Task] =
+      CostAccounting.unsafe[Task](Cost(initialPhlo))
+    implicit val costL                     = costLog[Task]
+    implicit val cost: _cost[Task]         = loggingCost(costAlg, costL)
+    val dbDir                              = Files.createTempDirectory("cost-accounting-spec-")
+    val size                               = 1024L * 1024 * 1024
+    implicit val logF: Log[Task]           = new Log.NOPLog[Task]
+    implicit val metricsEff: Metrics[Task] = new metrics.Metrics.MetricsNOP[Task]
+
+    (for {
+      costsLoggingProgram <- costL.listen({
+                              for {
+                                runtime <- Runtime
+                                            .create[Task, Task.Par](dbDir, size, StoreType.LMDB)
+                                par <- Interpreter[Task].buildNormalizedTerm(contract)
+                                res <- Interpreter[Task]
+                                        .evaluate(runtime, par, Cost(initialPhlo.toLong))
+                                _ <- runtime.close()
+                                _ <- Task.eval(dbDir.recursivelyDelete())
+                              } yield (res)
+                            }.attempt)
+      (result, costLog) = costsLoggingProgram
+      res               <- errorLog.readAndClearErrorVector
+      _                 = Task.now(res should be(Vector.empty))
+    } yield ((result, costLog))).unsafeRunSync
+  }
+
+  case class ContractWithCost(contract: String, cost: Long)
+
+  val shortslow = ContractWithCost(
+    """new loop in {
+         contract loop(@n) = {
+           match n {
+             0 => Nil
+             _ => loop!(n-1)
+           }
+         } |
+         loop!(10)
+       }""".stripMargin,
+    1766
```

Comment:
I'd like to see more ContractWithCost-s, and a test that uses contracts that have varying top-level par type (send / receive / new / expr /bundle / connective ...). While the tests in here look quite torough, they seem to be never testing a case when more than one `evalJob` is being run in parallel.

Reply:
I'm going to add one more case, but I looking forward to testing this with the par generator that is currently being implemented